Automattic/jetpack
# Search: Remove dependency on @wordpress/interface

Fixes #21065.

#### Changes proposed in this Pull Request:
* Removes `@wordpress/interface`.
* Removes usage of the interface package in Customberg.
* The Customberg sidebar is no longer dismissable in desktop breakpoints.
* The Customberg sidebar is now open by default on mobile breakpoints.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your Jetpack site with a Search subscription.
* Open Customberg (`/wp-admin/admin.php?page=jetpack-search-configure`).
* Ensure that the sidebar is always open on desktop viewports.
* Resize the window to mimic a mobile breakpoint and refresh the page.
* Ensure that the page opens with the sidebar open by default.
